test/cql-pytest/test_null.py

```diff
@@ -65,6 +79,51 @@ def test_insert_null_key_lwt(cql, table1):
     with pytest.raises(InvalidRequest, match='null value'):
         cql.execute(stmt, [None, s])
 
+# INSERT statements must specify all components of the primary key,
+# and the corresponding query parameters, if any, must not be null.
+# If INSERT or DELETE statement is missing a non-last element of
+# the primary key, the error message generated contained
+# an invalid column name (#12046).
+# The problem occurred if the query contains a column with the list type,
```

Comment:
Thanks. The tests look good, but I'm still not sure I understand why these tests need to use "c" as a list. If I understand correctly, the tests revolve around missing clustering key or partition key parts - and those are all "text" columns in your tests, not lists. Why does the type of C matter?

More on this below.

Reply:
The reason is dictated by the current structure of the code. If the statement contains list columns, the `modification_statement::_selects_a_collection` variable will be `true`, in `modification_statement::process_where_clause` it will be passed to `statement_restrictions::statement_restrictions` as `bool for_view = true` (this looks strange and seems like a bug, but nevertheless), then in `statement_restrictions::process_clustering_columns_restrictions`, since `for_view==false`, we will skip the check that all the components of the clustering key are set. So, the problem occurs only if the query contains a list column.

Reply:
Wow. Ok. This "for_view = true" thing looks very suspicious but I don't know if you still have the energy to continue chasing this (the list of commit grew to 6, can now continue to grow to 7 ;-) ).

Reply:
I'll look at the history of this "for_view = _selects_a_collection" thing and make another PR if I find something interesting)